pkg/api/types.go

```diff
@@ -624,36 +624,13 @@ const (
 	ClaimLost PersistentVolumeClaimPhase = "Lost"
 )
 
-type HostPathType string
-
-const (
-	// For backwards compatible, leave it empty if unset
-	HostPathUnset HostPathType = ""
-	// If nothing exists at the given path, an empty directory will be created there
-	// as needed with file mode 0755, having the same group and ownership with Kubelet.
-	HostPathDirectoryOrCreate HostPathType = "DirectoryOrCreate"
-	// A directory must exist at the given path
-	HostPathDirectory HostPathType = "Directory"
-	// If nothing exists at the given path, an empty file will be created there
-	// as needed with file mode 0644, having the same group and ownership with Kubelet.
-	HostPathFileOrCreate HostPathType = "FileOrCreate"
-	// A file must exist at the given path
-	HostPathFile HostPathType = "File"
-	// A UNIX socket must exist at the given path
-	HostPathSocket HostPathType = "Socket"
-	// A character device must exist at the given path
-	HostPathCharDev HostPathType = "CharDevice"
-	// A block device must exist at the given path
-	HostPathBlockDev HostPathType = "BlockDevice"
-)
-
 // Represents a host path mapped into a pod.
 // Host path volumes do not support ownership management or SELinux relabeling.
 type HostPathVolumeSource struct {
 	// If the path is a symlink, it will follow the link to the real path.
 	Path string
 	// Defaults to ""
-	Type *HostPathType
+	Type *metav1.FileType
```

Comment:
You can't refer to mount package from types.go, it's our public API. You can do it the other way around (mount pkg importing types.go) if you want.

Please revert any API changes, they're set to stone now.

Reply:
@jsafrane These have already moved this to `k8s.io/apimachinery`. Please see last commit.

As https://github.com/kubernetes/kubernetes/pull/51771#discussion_r137924250 stated, we need to refer `FileType` in `pkg/util/mount`.
> In order to avoid import cycle, we should find a place to declare these const vars. 

I think `k8s.io/apimachinery` is a good candidate. WDYT?

Reply:
It does not belong to API machinery. Don't change any `types.go` please. FileType is internal Kubernetes type and we don't want to expose it in public API.

`host_path.go` imports `pkg/util/mount`. So why don't you declare the constants in `pkg/util/mount/mount.go`? Would there be an import cycle that I don't see?

Reply:
Initially those const values are copied from [HostPathType](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L738-L743).

> host_path.go imports pkg/util/mount. So why don't you declare the constants in pkg/util/mount/mount.go? Would there be an import cycle that I don't see?

@jsafrane So we need to declare another same set of const values? Or we will face import cycle when importing `types.go`.